retrofit-mock/src/main/java/retrofit/mock/BehaviorCall.java

```diff
@@ -0,0 +1,183 @@
+  }
+
+  @Override public void enqueue(final Callback<T> callback) {
+    synchronized (this) {
+      if (executed) throw new IllegalStateException("Already executed");
+      executed = true;
+    }
+    task = backgroundExecutor.submit(new Runnable() {
+      @Override public void run() {
+        if (behavior.calculateIsFailure()) {
+          try {
+            Thread.sleep(behavior.calculateDelay(MILLISECONDS));
+          } catch (InterruptedException e) {
+            callbackExecutor.execute(new Runnable() {
+              @Override public void run() {
+                callback.onFailure(new InterruptedIOException("canceled"));
+              }
+            });
+            return;
+          }
+          callbackExecutor.execute(new Runnable() {
+            @Override public void run() {
+              callback.onFailure(behavior.failureException());
+            }
+          });
+        } else {
+          delegate.enqueue(new Callback<T>() {
+            @Override public void onResponse(final Response<T> response) {
+              long sleepMs = behavior.calculateDelay(MILLISECONDS);
+              if (sleepMs > 0) {
+                try {
+                  Thread.sleep(sleepMs);
+                } catch (InterruptedException e) {
+                  callbackExecutor.execute(new Runnable() {
+                    @Override public void run() {
+                      callback.onFailure(new InterruptedIOException("canceled"));
+                    }
+                  });
+                  return;
+                }
+              }
+              callbackExecutor.execute(new Runnable() {
+                @Override public void run() {
+                  callback.onResponse(response);
+                }
+              });
+            }
+
+            @Override public void onFailure(final Throwable t) {
+              long sleepMs = behavior.calculateDelay(MILLISECONDS);
+              if (sleepMs > 0) {
+                try {
+                  Thread.sleep(sleepMs);
+                } catch (InterruptedException e) {
+                  callbackExecutor.execute(new Runnable() {
+                    @Override public void run() {
+                      callback.onFailure(new InterruptedIOException("canceled"));
+                    }
+                  });
+                  return;
+                }
+              }
+              callbackExecutor.execute(new Runnable() {
+                @Override public void run() {
+                  callback.onFailure(t);
+                }
+              });
+            }
+          });
+        }
+      }
+    });
+  }
+
+  @Override public Response<T> execute() throws IOException {
+    synchronized (this) {
+      if (executed) throw new IllegalStateException("Already executed");
+      executed = true;
+    }
+    if (canceled) {
+      throw new InterruptedIOException("canceled");
+    }
+
+    Future<Response<T>> task = backgroundExecutor.submit(new Callable<Response<T>>() {
```

Comment:
If you're going to burn a thread anyway, I'd be tempted to have this call through the async path instead.